### PR TITLE
Additional reason for Customer's partial payment - bank charge

### PR DESCRIPTION
**Pull request by henrynopo**
For international bank transfer, there is often some charge by the intermediate bank and this amount is deducted directly from the amount received.

I'd like to propose adding an additional reason (namely "bank charge") for classifying partial payment of customer's invoice. To achieve this, it still need to add in the translation as below:
- en_US or en_GB	bankcharge	Bank Charge	   
- en_US or en_GB	ConfirmClassifyPaidPartiallyReasonBankCharge	Bank Charge	   
- en_US or en_GB	ConfirmClassifyPaidPartiallyReasonBankChargeDesc	The difference was charged by bank while Client paid the correct amount.

### CODE REVIEW

File: htdocs/core/class/html.form.class.php [hunk 1]
```diff
@@ -1511,8 +1511,8 @@ public function select_remises($selected, $htmlname, $filter, $socid, $maxvalue
 		global $langs, $conf;
 
 		// On recherche les remises
-		$sql = "SELECT re.rowid, re.amount_ht, re.amount_tva, re.amount_ttc,";
-		$sql .= " re.description, re.fk_facture_source";
+		$sql = "SELECT re.rowid, re.amount_ht, re.amount_tva, re.amount_ttc, re.multicurrency_amount_ttc,";
+		$sql .= " re.description, re.fk_facture_sourcere, re.fk_invoice_supplier_source";
```
Comment (eldy) [inline htdocs/core/class/html.form.class.php:1515]: Field fk_facture_sourcere seems wrong

File: htdocs/core/class/html.form.class.php [hunk 2]
```diff
@@ -1563,10 +1563,21 @@ public function select_remises($selected, $htmlname, $filter, $socid, $maxvalue
 						$tmpfac = new Facture($this->db);
 						if ($tmpfac->fetch($obj->fk_facture_source) > 0) {
 							$desc = $desc.' - '.$tmpfac->ref;
+							$multicurrency_code = $tmpfac->multicurrency_code;
 						}
 					}
+					// show Source Supplier Invoice Ref
+					else {
+        				if (!empty($conf->global->MAIN_SHOW_FACNUMBER_IN_DISCOUNT_LIST) && !empty($obj->fk_invoice_supplier_source)) {
+							$tmpfac = new FactureFournisseur($this->db);
```
Comment (eldy) [inline htdocs/core/class/html.form.class.php:1572]: Tabulation must use tab, not spaces.

File: htdocs/core/class/html.form.class.php (hunk 1 above)
Comment (henrynopo) [reply htdocs/core/class/html.form.class.php:1515]: You are right. some typo error, as I am new to Github.

File: htdocs/core/class/html.form.class.php (hunk 2 above)
Comment (henrynopo) [reply htdocs/core/class/html.form.class.php:1572]: Thanks. not a programmer, I am trying my best to understand the dolibarr code and tweak it to express my humble suggestions.

**Pull request closed without merging**